api/resource.rb

```diff
@@ -53,6 +53,11 @@ module Properties
       # list.  Otherwise, it's safe to leave empty.
       # If empty, we assume that `name` is the identifier.
       attr_reader :identity
+      # This is useful in case you need to change the query made for
+      # GET requests only. In particular, this is often used
+      # to extract an object from a parent object or a collection.
+      attr_reader :nested_decoder
```

Comment:
What about something like `list_on_get`? I'm personally not a huge fan of `nested_decoder`

Reply:
You know, it's not a great name but I feel very strongly that `list_on_get` is not a great replacement. I named it that way because it's a standarized decoder for nested objects (see custom_code.decoder for terraform), i.e. it takes a parent resource --> nested objects. I also came up with `nested_extractor` or just `extractor`?

Reply:
I wasn't a huge fan of decoder because we use it in a completely different way (encoders/decoders). 

I think only Ansible uses that function of Magic Modules, so it should really be removed.

Reply:
no, we use custom encoders and decoders in Terraform and that's why I named it so - because this essentially gets turned into a custom decoder that several fine-grained resources would use. 

We could also do something like `nested_query` since this would replace self-link query for us.